src/main/java/Duke.java

```diff
@@ -1,10 +1,148 @@
-        System.out.println("Hello from\n" + logo);
+
+    private TaskList tasks;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Constructor for Duke to start the initialise the necessary variables
+     */
+    public Duke() {
+        storage = new Storage();
+
+
+        try {
+            tasks = storage.load();
+            ui = new Ui(tasks);
+
+        } catch (FileNotFoundException e) {
+            System.out.println("NONEEE");
+            tasks = new TaskList();
+            ui = new Ui(tasks);
+        }
+
+    }
+
+
+    /**
+     * Starts the whole program and get user input
+     * @throws IOException if I/O error happens
+     */
+    public void run() throws IOException {
+        ui.printIntro();
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        while (!input.equals("bye")) {
+
+            Parser parser = new Parser();
+            parser.parse(input, ui, tasks);
+            storage.updateData(tasks);
+
+            input = sc.nextLine();
+        }
+
+
+        ui.printGoodbye();
+
+    }
+
+
+    /**
+     * Main method for the class
+     * @param args no commmand line arguments are used explicitly
+     * @throws IOException is thrown when there is I/O error
+     */
+    public static void main(String[] args) throws IOException {
+        new Duke().run();
+    }
+
+}
+
+/**
+ * Task class to store task information
+ */
+class Task {
+    private String description;
+    private boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+}
+
+
+/**
+ * Deadline class which is a Task but more specific
+ */
+class Deadline extends Task {
+    private LocalDate by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+}
+
+/**
+ * Todo class which is a Task but more specific
+ */
+class Todo extends Task {
+    public Todo(String description) {
+        super(description);
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+}
+
+/**
+ * Event class which is a Task but more specific
+ */
+class Event extends Task {
+    private String at;
+
+    public Event(String description, String at) {
+        super(description);
+        this.at = at;
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + at + ")";
     }
 }
+
```

Comment:
Best to have only one class per file, even for subclasses! It makes things easier (especially in the case of collaborators) to find a piece of code by just looking for the file name that contains the class name, instead of starting a manhunt, especially if the project becomes very complex!

Javadocs for the task subclasses are also a bit vague, maybe describe a bit more on what 'specific' is?